### PR TITLE
[SPARK-38352][SQL] Fix DataFrameAggregateSuite/DataFrameSetOperationsSuite/DataFrameWindowFunctionsSuite under ANSI mode

### PR DESCRIPTION
**Pull request by gengliangwang**
### What changes were proposed in this pull request?

Fix the following test suites under ANSI mode:

- DataFrameAggregateSuite
- DataFrameNaFunctionsSuite
- DataFrameRangeSuite
- DataFrameSetOperationsSuite
- DataFrameWindowFunctionsSuite
- TwoLevelAggregateHashMapSuite
- TwoLevelAggregateHashMapWithVectorizedMapSuite

### Why are the changes needed?

To set up a new GA test job with ANSI mode on

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

Manually turn on ANSI mode and test .
Also it should pass GA tests.

### CODE REVIEW

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameNaFunctionsSuite.scala [hunk 1]
```diff
@@ -444,21 +444,25 @@ class DataFrameNaFunctionsSuite extends QueryTest with SharedSparkSession {
   }
 
   test("replace float with nan") {
-    checkAnswer(
-      createNaNDF().na.replace("*", Map(
-        1.0f -> Float.NaN
-      )),
-      Row(0, 0L, 0.toShort, 0.toByte, Float.NaN, Double.NaN) ::
-      Row(0, 0L, 0.toShort, 0.toByte, Float.NaN, Double.NaN) :: Nil)
+    if (!conf.ansiEnabled) {
```
Comment (cloud-fan) [inline sql/core/src/test/scala/org/apache/spark/sql/DataFrameNaFunctionsSuite.scala:447]: how does ansi break the test?

Comment (gengliangwang) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameNaFunctionsSuite.scala:447]: yeah, this case casts Float.NaN as Int.
For non-ansi, the result is 0

File: sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala [hunk 2]
```diff
@@ -97,7 +97,8 @@ class DataFrameWindowFunctionsSuite extends QueryTest
   }
 
   test("corr, covar_pop, stddev_pop functions in specific window") {
-    withSQLConf(SQLConf.LEGACY_STATISTICAL_AGGREGATE.key -> "true") {
+    withSQLConf(SQLConf.LEGACY_STATISTICAL_AGGREGATE.key -> "true",
+      SQLConf.ANSI_ENABLED.key -> "false") {
```
[the remark this answers is not in the document]
Comment (gengliangwang) [reply sql/core/src/test/scala/org/apache/spark/sql/DataFrameWindowFunctionsSuite.scala:101]: divide by 0

**Review by cloud-fan — APPROVED** [review]

**Comment by gengliangwang** [comment]
Merging to master

**Pull request closed without merging**